dinstaller-lib/src/store.rs

```diff
@@ -37,18 +41,40 @@ impl<'a> Store<'a> {
     }
 
     /// Stores the given installation settings in the D-Bus service
-    pub fn store(&self, settings: &Settings) -> Result<(), Box<dyn Error>> {
+    pub fn store(&self, settings: &InstallSettings) -> Result<(), Box<dyn Error>> {
+        dbg!("Storing {}", &settings);
+
+        self.store_software_settings(&settings.software)?;
+        self.store_user_settings(&settings.user)?;
+        self.store_storage_settings(&settings.storage)?;
+
+        Ok(())
+    }
+
+    fn store_user_settings(&self, settings: &UserSettings) -> Result<(), Box<dyn Error>> {
         // fixme: improve
         let first_user = FirstUser {
-            user_name: settings.user.user_name.clone(),
-            full_name: settings.user.full_name.clone(),
-            autologin: settings.user.autologin,
-            password: settings.user.password.clone(),
+            user_name: settings.user_name.clone(),
+            full_name: settings.full_name.clone(),
+            autologin: settings.autologin,
+            password: settings.password.clone(),
             ..Default::default()
         };
-        self.software_client
-            .select_product(&settings.software.product)?;
         self.users_client.set_first_user(&first_user)?;
         Ok(())
     }
+
+    fn store_software_settings(&self, settings: &SoftwareSettings) -> Result<(), Box<dyn Error>> {
+        self.software_client.select_product(&settings.product)?;
+        Ok(())
+    }
+
+    fn store_storage_settings(&self, settings: &StorageSettings) -> Result<(), Box<dyn Error>> {
+        self.storage_client.calculate(
+            settings.devices.iter().map(|d| d.name.clone()).collect(),
+            settings.encryption_password.clone(),
+            settings.lvm,
+        )?;
+        Ok(())
```

Comment:
Here I think we should also report if calculate failed? ( at least add todo )

Reply:
OK, I will add a todo.

Reply:
I have added a note. Thanks!